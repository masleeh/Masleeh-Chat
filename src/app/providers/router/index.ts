import AppRouter from "./ui/AppRouter";
import { AppRouterProvider, router as AppBrowserRouter } from "./ui/RouterProvider";

export {
    AppRouter,
    AppRouterProvider,
    AppBrowserRouter
}