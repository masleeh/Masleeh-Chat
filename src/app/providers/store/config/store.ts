import { ReducersMapObject, configureStore } from "@reduxjs/toolkit"
import { IStateSchema } from "./StateSchema"
import { userReducer } from "entities/User"


export const createReduxStore = (initialState?: IStateSchema) => {
    const rootReducers: ReducersMapObject<IStateSchema> = {
        user: userReducer
    }

    return configureStore<IStateSchema>({
        reducer: rootReducers,
        devTools: true,
        preloadedState: initialState
    })
}

export type RootState = ReturnType<typeof createReduxStore>['getState']
export type AppDispatch = ReturnType<typeof createReduxStore>['dispatch']