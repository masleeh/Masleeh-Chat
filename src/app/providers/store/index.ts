import StoreProvider from "./StoreProvider";
import type { IStateSchema } from "./config/StateSchema";

export {
    StoreProvider
}

export type {
    IStateSchema
}