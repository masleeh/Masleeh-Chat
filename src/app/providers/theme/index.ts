import AppThemeProvider from './ui/ThemeProvider'

export {
    AppThemeProvider
}