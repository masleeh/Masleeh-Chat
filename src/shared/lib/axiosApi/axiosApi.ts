import axios from 'axios'

const $api = axios.create({
    withCredentials: true,
    baseURL: import.meta.env.VITE_API_URL
})

$api.interceptors.request.use((config) => {
    config.headers.Authorization = `Bearer ${localStorage.getItem(import.meta.env.VITE_LOCALSTORAGE_TOKEN_KEY)}`
    return config
})

$api.interceptors.response.use((config) => {
    return config   
}, async (error) => {
    const originalRequest = error.config;
    if (error.response.status === 401 && error.config && !error.config._isRetry) {
        originalRequest._isRetry = true
        try {
            const response = await axios.get(`${import.meta.env.VITE_API_URL}/auth/refresh`, { withCredentials: true })
            localStorage.setItem(import.meta.env.VITE_LOCALSTORAGE_TOKEN_KEY, response.data.access_token)
            return $api.request(originalRequest)
        } catch (error) {
            console.log("Not authorized")
        }
    }
    throw error
})

export default $api