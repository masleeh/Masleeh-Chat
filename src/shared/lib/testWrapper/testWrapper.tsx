import { render } from "@testing-library/react"
import { IStateSchema, TestStoreProvider } from "app/providers/store";
import i18nTest from 'shared/config/i18n/i18nTest'
import { I18nextProvider } from "react-i18next"
import { MemoryRouter } from "react-router-dom";
import { DeepPartial, EnhancedStore } from "@reduxjs/toolkit";
import { AppThemeProvider } from "app/providers/theme";

export interface componentRenderOptions {
    route?: string;
    initialState?: DeepPartial<IStateSchema>;
    store?: EnhancedStore
}

const testWrapper = (component: React.ReactNode, options: componentRenderOptions = {}) => {
    const {
        route = '/',
        initialState,
        store
    } = options

    return render(
        <TestStoreProvider initialState={initialState} store={store}>
            <AppThemeProvider>
                <MemoryRouter initialEntries={[route]}>
                    <I18nextProvider i18n={i18nTest}>
                        {component}
                    </I18nextProvider>
                </MemoryRouter>
            </AppThemeProvider>
        </TestStoreProvider>
    )
}

export default testWrapper