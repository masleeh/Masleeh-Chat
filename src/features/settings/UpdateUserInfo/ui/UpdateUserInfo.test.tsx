import {screen} from '@testing-library/react'
import testWrapper from "shared/lib/testWrapper/testWrapper"
import { UpdateUserInfo } from '..'
import userEvent from '@testing-library/user-event'
import { DeepPartial } from 'react-hook-form'
import $api from 'shared/lib/axiosApi/axiosApi'
import { IStateSchema, createReduxStore } from 'app/providers/store'

const axiosMock = vi.spyOn($api, 'patch')
const state: DeepPartial<IStateSchema> = {
    user: {
        userData: {
            username: 'masleeh',
            user_id: 'fowe8vke7c;wh',
            profile_pic: 'http://localhost:5000/users/Ea3bptxnJF2-dqp6Wk8SHEZOlVkw_oyaJz4QlO6h.jpg'
        }
    }
}

const axiosResponse = {
    username: 'kunteynir',
    user_id: 'fowe8vke7c;wh',
    profile_pic: 'http://localhost:5000/users/Ea3bptxnJF2-dqp6Wk8SHEZOlVkw_oyaJz4QlO6h.jpg'
}

describe('UpdateUserInfo component', () => {
    it('Check UpdateUserInfo', async () => {
        const component = testWrapper(<UpdateUserInfo />)
        const showButton = screen.getByLabelText("s show user info dialog")
        await userEvent.click(showButton)
        expect(component).toMatchSnapshot()
    })

    it('Check full functional', async () => {
        axiosMock.mockResolvedValue(axiosResponse)
        const store = createReduxStore(state as IStateSchema)
        testWrapper(<UpdateUserInfo />, {store: store})
        const showButton = screen.getByLabelText("s show user info dialog")
        await userEvent.click(showButton)

        const usernameInput = screen.getByLabelText("s username input")
        await userEvent.type(usernameInput, 'kunteynir')

        const updateButton = screen.getByLabelText("update user info button")
        await userEvent.click(updateButton)

        setTimeout(() => {
            const newState = store.getState()
            expect(newState.user.userData).toEqual(axiosResponse)
        }, 1100)
    })

})