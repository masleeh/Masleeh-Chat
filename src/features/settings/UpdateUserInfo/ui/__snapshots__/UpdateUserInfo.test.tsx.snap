// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UpdateUserInfo component > Check UpdateUserInfo 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    >
      <button
        aria-label="s show user info dialog"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        s_edit
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        >
          <span
            class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
            style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
          >
            <span
              class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
            />
          </span>
        </span>
      </button>
    </div>
    <form
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r0:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-12navqv-MuiTypography-root-MuiDialogTitle-root"
            id=":r0:"
          >
            s_edit
          </h2>
          <div
            class="MuiDialogContent-root css-y3n1w3-MuiDialogContent-root"
          >
            <div
              aria-label="s username input"
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-nxjks2-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard MuiFormLabel-colorPrimary Mui-focused MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="s_username"
                id="s_username-label"
              >
                s_username
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-l4u8b9-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  id="s_username"
                  name="username"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-11uq76c-MuiCollapse-root"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                >
                  <div
                    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-e9aaga-MuiPaper-root-MuiAlert-root"
                    role="alert"
                  >
                    <div
                      class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                        data-testid="ErrorOutlineIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
                    />
                    <div
                      class="MuiAlert-action css-ki1hdl-MuiAlert-action"
                    >
                      <button
                        aria-label="close"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                          data-testid="CloseIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing css-i4drsf-MuiDialogActions-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeMedium MuiButton-textSizeMedium css-10fi992-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              s_cancel
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="update user info button"
              class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-u25v72-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
              id=":r2:"
              tabindex="0"
              type="submit"
            >
              s_save
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </form>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <button
      aria-label="s show user info dialog"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-11qr2p8-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      s_edit
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      >
        <span
          class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
